naga/src/back/msl/writer.rs

```diff
@@ -611,6 +611,13 @@ trait NameKeyExt {
             FunctionOrigin::EntryPoint(idx) => NameKey::EntryPointLocal(idx, local_handle),
         }
     }
+
+    fn oob_local(origin: FunctionOrigin, ty: Handle<crate::Type>) -> NameKey {
```

Comment:
question: It's rather confusing to only use the noun `local` when there are _types_ at play here rather than just local variables.

suggestion: Let's rename to something like `s/local/local_type`.

Reply:
`oob_local_for_type` seems like a good option to me!